sunpy/map/mapcube.py

```diff
@@ -360,3 +361,17 @@ def all_meta(self):
         Return all the meta objects as a list.
         """
         return [m.meta for m in self.maps]
+
+    def save(self, prefix_path):
+        """
+        Saves the Maps held by the MapCube object in a directory.
+
+        Parameters
+        ----------
+        prefix_path : str
+            /path/to/directory where file should be saved.
+        """
+        for index, _map in enumerate(self.maps):
+            filename = "%d.fits"%(index)
```

Comment:
It is important that when the files are listed the list follows strict numerical ordering.  This normally means adding in the requisite number of leading zeroes to the index.  So if there are 1200 files in the mapcube, then the first file is saved as with "0000", the second file "0001", the 50th file is "0049", the 756th file is "0755" and so on.  Looks like https://stackoverflow.com/questions/29000613/add-leading-zeros-to-a-list-of-numbers-in-python can help with this.

Reply:
Also using the new `.format` syntax it's needed here.

Reply:
Since I feel there are conflicting things in this thread, let me confirm once:
1. I find the notation quite neat and sensible, have you considered need to scan it back in, say a `load(filepath)`. Also, I didn't quite understand the `file.<index>.` prefix notation - can you please clarify?
2. `.format` 👍 
3. Should I extend filetype to mapcube save, like @Punyaslok said?
4. Do we have `__eq__` on `GenericMap` on which I can run reduce for `MapCube`? I found one test in [sunpy/map/tests/test_map_factory.py](https://github.com/sunpy/sunpy/blob/master/sunpy/map/tests/test_map_factory.py#L104-L110), which I've used to create a similar test and added to this PR.

Reply:
(1a) I think the prefix_path is fine as it is.  Map can read in a list of file names to get a cube back so there is no need for a load method.

(1b) The reason for the zeroes formatting is due to the way Linux and other OSs order file names when listed.  Consider what happens when you have 20 maps in a mapcube. If you were to use the current code and do an "ls" on the directory where the files are, then according to the way Linux lists files the list would be ordered as 

file.0.fits, file.1.fits, file.11.fits, file.12.fits,.... 

and so on.  This is not really very useful, as it destroys the order of the files as intended.  If you put in enough leading zeroes, then the file listing comes out with the order as intended - file.00.fits, file.01.fits, file.02.fits, .....

(3) Yes, that's a good idea, as this ensures changes to how we save maps get passed through to mapcube too.

(4) There is no __eq__ on GenericMap.  I think the test you have should test that the order of the maps in the mapcube you read back in is the same as the original order.